frontend/src/metabase/components/SaveQuestionForm/context.tsx

```diff
@@ -68,6 +70,12 @@ export const SaveQuestionProvider = ({
     originalQuestion?.collectionId(),
   );
 
+  const currentUser = useSelector(getCurrentUser);
+  const targetCollection =
+    currentUser && saveToCollection === "personal"
+      ? currentUser.personal_collection_id
+      : saveToCollection;
```

Comment:
I was thinking of renaming this to `targetCollectionId` as we already casted `"personal"` to the personal_collection_id, but I think we also support `"root"` so we can leave this as-is 👍🏻